File: backend/entity/payment_slip.go
// entity/payment_slip.go
package entity

import (
	"time"

	"gorm.io/gorm"
)

type PaymentSlip struct {
	gorm.Model
	UploadAt time.Time `json:"upload_at"`
	FileURL  string    `json:"file_url"`

	PaymentID uint     `json:"payment_id"`
	Payment   *Payment `gorm:"foreignKey:PaymentID" json:"payment,omitempty"`
}

File: frontend/src/interfaces/PaymentSlip.ts
import type { Payment } from "./Payment";

export interface PaymentSlip {
  ID: number;
  upload_at: string; // ISO datetime
  file_url: string;

  payment_id: number;
  payment?: Payment;
}

export interface CreatePaymentSlipRequest {
  upload_at?: string;
  file_url: string;
  payment_id: number;
}
